### PR TITLE
feat: remote callbacks

### PR DESCRIPTION
**Pull request by SankaD**
callback response object is received

### CODE REVIEW

File: lib/saito/peer.js [hunk 1]
```diff
@@ -214,81 +215,84 @@ console.log("done emitting handshake complete");
...
+                break;
+            default:
+
+
+                //
+                // if the attached data is a transaction, ensure MSG field exists
+                //
+                // transaction.msg included for backwards compatibility
+                //
+                if (reconstructed_data) {
```
Comment (escapedcat) [inline lib/saito/peer.js:279]: Shouldn't [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) work here?:

`if (reconstructed_data?.transaction?.m) {`

File: lib/saito/peer.js [hunk 2]
```diff
@@ -404,24 +408,29 @@ console.log("old keyist request");
...
+    //     console.log(data);
+    //     if (this.socket && this.socket.readyState === this.socket.OPEN) {
+    //         await this.app.networkApi.sendAPIResponse(this.socket, "RESULT__", message_id, Buffer.from(data, 'utf-8'));
+    //     } else {
+    //         console.error("socket not open");
+    //         console.log(this);
+    //     }
+    // }
+    async sendResponse(message_id, data) {
+        console.log("sendResponse");
```
Comment (escapedcat) [inline lib/saito/peer.js:422]: Could save a log by doing `console.log("sendResponse", data);`

File: lib/saito/peer.js (hunk 1 above)
Comment (SankaD) [reply lib/saito/peer.js:279]: only formatted the current code here.

File: lib/saito/peer.js (hunk 2 above)
Comment (SankaD) [reply lib/saito/peer.js:422]: thanks.  actually this log will be probably removed.

**Pull request merged**